code/processes/chainedtp.q

```diff
@@ -196,13 +186,15 @@ if[not `upd in key `.; upd:.ctp.upd];
 /- pubsub must be initialised sooner to enable tickerplant replay publishing to work
 .ps.initialise[];                                                                   
 
-/- check if the tickerplant has connected, blocks the process until a connection is established
-.ctp.initialise[];
-while[.ctp.notpconnected[];
-  /- while not connected make the process sleep for X seconds and then run the subscribe function again
-  .os.sleep[.ctp.tpconnsleep];
-  /- start chained tickerplant
-  .ctp.initialise[]];
+/- check if tickerplant is available and if not exit with error 
+.servers.startupdepcycles[.ctp.tickerplanttypes;.ctp.tpconnsleep;.ctp.tpcheckcycles];
```

Comment:
I think this could cause problems in systems with multiple tickerplants. Say for example we have a system with 2 TPs, named `tickerplant1` & `tickerplant2`, both proctype `tickerplant`.

If we then have two chained TPs, one for each regular TP, this line which connects based on _proctype_ will only ensure that ctp is connected to one of the TPs, but not necessarily the right one.

e.g. if only `tickerplant1` is running when `chainedtp1` & `chainedtp2` start, I think they'll both connect to it, instead of `chainedtp2` waiting for `tickerplant2`. I think the code in the subscribe function will stop it actually subscribing to the wrong TP, but there's now no retry logic, so `chainedtp2` in this scenario would never subscribe to `tickerplant2` when it does come up